pandas/core/frame.py

```diff
@@ -3354,38 +3354,23 @@ def assign(self, **kwargs):
 
         Examples
         --------
-        >>> df = pd.DataFrame({'A': range(1, 11), 'B': np.random.randn(10)})
+        >>> df = pd.DataFrame({'temp_c': (17.0, 25.0)},
+                                index=['portland', 'berkeley'])
 
         Where the value is a callable, evaluated on `df`:
 
-        >>> df.assign(ln_A = lambda x: np.log(x.A))
-            A         B      ln_A
-        0   1  0.426905  0.000000
-        1   2 -0.780949  0.693147
-        2   3 -0.418711  1.098612
-        3   4 -0.269708  1.386294
-        4   5 -0.274002  1.609438
-        5   6 -0.500792  1.791759
-        6   7  1.649697  1.945910
-        7   8 -1.495604  2.079442
-        8   9  0.549296  2.197225
-        9  10 -0.758542  2.302585
+        >>> df.assign(temp_f=lambda x: x.temp_c * 9 / 5 + 32)
+                  temp_c  temp_f
+        portland    17.0    62.6
+        berkeley    25.0    77.0
 
         Where the value already exists and is inserted:
 
-        >>> newcol = np.log(df['A'])
-        >>> df.assign(ln_A=newcol)
-            A         B      ln_A
-        0   1  0.426905  0.000000
-        1   2 -0.780949  0.693147
-        2   3 -0.418711  1.098612
-        3   4 -0.269708  1.386294
-        4   5 -0.274002  1.609438
-        5   6 -0.500792  1.791759
-        6   7  1.649697  1.945910
-        7   8 -1.495604  2.079442
-        8   9  0.549296  2.197225
-        9  10 -0.758542  2.302585
+        >>> newcol = df['temp_c'] * 9 / 5 + 32
+        >>> df.assign(temp_f=newcol)
+                  temp_c  temp_f
+        portland    17.0    62.6
+        berkeley    25.0    77.0
 
         Where the keyword arguments depend on each other
```

Comment:
I think this last example is not useful anymore, or do you think it is?

Reply:
This example is still necessary in some form, as it shows a non-obvious Python 3.6+ feature that was recently added (#18852).  It could definitely be improved though.

Reversing the setup and starting from a DataFrame with Fahrenheit data:
```python
In [2]: df = pd.DataFrame({'temp_f': [-40, 0, 32, 100, 212]})

In [3]: df
Out[3]:
   temp_f
0     -40
1       0
2      32
3     100
4     212
```

Then what this is section is trying to illustrate is that, in Python 3.6+, you can create multiple columns within the same `assign` where one of the columns depends on another one defined within the same `assign`.

For example, if you wanted to add both Celsius and Kelvin columns, you could use a formula to build Celsius from Fahrenheit, then use the Celsius column to create the Kelvin one (instead of using a more complex formula referencing Fahrenheit), all from within the same `assign`:
```python
In [4]: df.assign(temp_c=lambda df: 5 / 9 * (df['temp_f'] - 32), 
                  temp_k=lambda df: df['temp_c'] + 273.15)
Out[4]:
   temp_f      temp_c      temp_k
0     -40  -40.000000  233.150000
1       0  -17.777778  255.372222
2      32    0.000000  273.150000
3     100   37.777778  310.927778
4     212  100.000000  373.150000
```

There have also been some more generic questions about how to add multiple columns using `assign` (e.g. [StackOverflow](https://stackoverflow.com/q/42101382/3339965)).  It might be beneficial to first give an example similar to the one above, but where you create Kelvin from Fahrenheit, to simply show how to add multiple columns within the same `assign`.  Then the 3.6+ behavior could be discussed using something similar to my example above.

Reply:
Ah! I understand now why this example is needed. Let me add a different one that better portrays its usage.